agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/models/TelemetryItem.java

```diff
@@ -215,15 +220,34 @@ public String getInstrumentationKey() {
     return this.instrumentationKey;
   }
 
-  /**
-   * Set the instrumentationKey property: The instrumentation key of the Application Insights
-   * resource.
-   *
-   * @param instrumentationKey the instrumentationKey value to set.
-   * @return the TelemetryItem object itself.
-   */
-  public TelemetryItem setInstrumentationKey(String instrumentationKey) {
-    this.instrumentationKey = instrumentationKey;
+  @JsonIgnore
+  public String getConnectionString() {
+    return connectionString;
+  }
+
+  @JsonIgnore
+  public TelemetryItem setConnectionString(String connectionString) {
+    this.connectionString = connectionString;
+    this.instrumentationKey = ConnectionString.parse(connectionString).getInstrumentationKey();
+    return this;
+  }
+
+  @JsonIgnore
+  public TelemetryItem setConnectionString(ConnectionString connectionString) {
+    this.connectionString = connectionString.getParsedFrom();
+    this.instrumentationKey = connectionString.getInstrumentationKey();
+    return this;
+  }
+
+  @JsonIgnore
+  public TelemetryItem setConnectionString(StatsbeatConnectionString connectionString) {
+    instrumentationKey = connectionString.getInstrumentationKey();
+    // TODO (heya) turn StatsbeatConnectionString into a real connection string?
+    this.connectionString =
+        "InstrumentationKey="
+            + connectionString.getInstrumentationKey()
+            + ";IngestionEndpoint="
+            + connectionString.getIngestionEndpoint();
```

Comment:
```suggestion
    this.connectionString =
        "InstrumentationKey="
            + instrumentationKey
            + ";IngestionEndpoint="
            + connectionString.getIngestionEndpoint();
```

might just reuse line 244.